graspologic/cluster/autogmm.py

```diff
@@ -375,10 +390,12 @@ def __init__(
         self.selection_criteria = selection_criteria
         self.max_agglom_size = max_agglom_size
         self.n_jobs = n_jobs
+        self.n_init = n_init
+        self.run_multiple_init = run_multiple_init
 
     def _fit_cluster(self, X, X_subset, y, params, agg_clustering):
         label_init = self.label_init
-        if label_init is not None:
+        if label_init is not None and "n_init" not in set(params[1].keys()):
```

Comment:
I'm confused here. So in `_process_paramgrid` it seems that `n_init` is only updated into `gm_params_processed` if `run_multiple_init` is True which means that `label_init` is None. So what is the purpose for the second condition?

Reply:
Basically, if `label_init` is not None, the code will perform `gm_params["weights_init"] = weights_init`, `gm_params["means_init"] = means_init` and `gm_params["precisions_init"] = precisions_init` which would go against what we are trying to do with this merge.

Reply:
I am just trying to understand how the 2nd condition relates to the 1st one. To be specific, it's checking whether `n_init` is in `set(params[1].keys())`, but at line 723, `n_init` is added with a condition `run_multiple_init` being True (at line 720). However, at line 379, `run_multiple_init` is set True only if `label_init` is None (negation of 1st condition). 

Therefore,
1st condition met --> 'run_multiple_init' being False --> won't add `n_init` --> 2nd condition met
Am I understanding it correctly?

Reply:
Yes, since I wrote this a couple of weeks ago I am actually running again over my code to ensure this step is not redundant and in case it is not, to rewrite it in a more readable format.

Reply:
Thank you for finding such a redundant condition, I just removed it : ) I updated the verification Notebook to reflect the change and the output is untouched.